pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -2425,10 +2427,22 @@ public void repairSegmentsInErrorStateForPauselessConsumption(String realtimeTab
       }
     }
 
+    _controllerMetrics.setOrUpdateTableGauge(realtimeTableName,
+        ControllerGauge.PAUSELESS_SEGMENTS_IN_ERROR_COUNT, segmentsInErrorStateInAllReplicas.size());
+
     if (segmentsInErrorStateInAtLeastOneReplica.isEmpty()) {
       return;
     }
 
+    boolean isPartialUpsertEnabled =
```

Comment:
We want to move this after the warning log below. We should also log something because we are not able to fix the segments. Probably an error log

Reply:
Probably also emit a metric for error segments not able to fix. We need to get immediate alert on it

Reply:
Addressed